File: 0283-move-zeroes/0283-move-zeroes.go
func moveZeroes(nums []int)  {
    curr_idx := 0
    for i := 0; i < len(nums); i++ {
        if nums[i] != 0 {
            nums[curr_idx] = nums[i]
            curr_idx++
        }
    }
    
    for i := curr_idx; i < len(nums); i++ {
        nums[i] = 0
    }
}

File: 0283-move-zeroes/0283-move-zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int zero_counts = 0;
    int curr_idx = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] != 0) {
            nums[curr_idx++] = nums[i];
        } else {
            zero_counts++;
        }
    }
    
    for(int i = numsSize - zero_counts; i < numsSize; i++) {
        nums[i] = 0;
    }
}